### PR TITLE
Issue 38206: FilePathRoot in exp.Runs table gets mangled when sibling folders are moved

### PR DESCRIPTION
**Pull request by labkey-jeckels**
Address many additional issues:

Check destination path instead of source path for dir/file check, since move has already happened
Consolidate duplicated code
Don't match on exp.experimentrun name anymore since that's not really a path element
Consistently use correct slash format based on S3 vs local storage
Misc cleanup

### CODE REVIEW

File: api/src/org/labkey/api/files/TableUpdaterFileListener.java [hunk 1]
```diff
@@ -242,137 +235,102 @@ public void fileMoved(@NotNull Path src, @NotNull Path dest, @Nullable User user
...
-            }
-            LOG.info("Updated " + rows + " row in " + _table + " for move from " + src + " to " + dest);
+        int rows;
+        try
+        {
+            rows = new SqlExecutor(schema).execute(singleEntrySQL);
         }
+        catch (RuntimeSQLException x)
+        {
+            // Retry once if it's a deadlock-related exception and we're not inside a transaction
```
Comment (labkey-kevink) [inline api/src/org/labkey/api/files/TableUpdaterFileListener.java]: Maybe use Matt's new DbScope.executeWithRetry?

**Review by labkey-kevink — APPROVED** [review]

File: api/src/org/labkey/api/files/TableUpdaterFileListener.java (hunk 1 above)
Comment (labkey-matthewb) [reply api/src/org/labkey/api/files/TableUpdaterFileListener.java]: Definitely!

Comment (labkey-jeckels) [reply api/src/org/labkey/api/files/TableUpdaterFileListener.java]: OK. I'm not understanding what exceptions I'm supposed to throw though. It has to be a DeadlockLoserDataAccessException? Other usages seem to use RuntimeSQLException or SQLException.

I'm calling SqlExecutor.execute() which throws a RuntimeSQLException. Do I have to unwrap and then get Spring to translate to DeadlockLoserDataAccessException?

Comment (labkey-matthewb) [reply api/src/org/labkey/api/files/TableUpdaterFileListener.java]: Or we can expand retryWithException to catch and inspect more exceptions

Comment (labkey-jeckels) [reply api/src/org/labkey/api/files/TableUpdaterFileListener.java]: I just pushed a commit that I think should do that, please take a look.

Comment (labkey-matthewb) [reply api/src/org/labkey/api/files/TableUpdaterFileListener.java]: Looks right to me.

**Pull request closed without merging**